tests/test_typing/bad/err3.catala_en

```diff
@@ -12,5 +12,61 @@ scope S:
 
 ```catala-test-inline
 $ catala Typecheck
-[RESULT] Typechecking successful!
+[ERROR] Error during typechecking, incompatible types:
+--> integer
+--> decimal
+
+Error coming from typechecking the following expression:
+  --> tests/test_typing/bad/common.catala_en
+   | 
+15 |   output a content decimal
+   |          ^
+   + 
+
+Type integer coming from expression:
+  --> tests/test_typing/bad/err3.catala_en
+   | 
+10 |   definition a equals number of (z ++ z) / 2
+   |                       ^^^^^^
+   + 
+
+Type decimal coming from expression:
+  --> tests/test_typing/bad/common.catala_en
+   | 
+15 |   output a content decimal
+   |                    ^^^^^^^
+   + 
+#return code 255#
+```
+
+A bug with delayed typechecking can make the above accept the error while it is
```

Comment:
What is that ?

Reply:
Just a comment on why the test is needed. Should it rather be in a ``catala block ?

Reply:
Here we get the same error twice, which is correct behaviour. If you look at the sate before the fix the test above answers "typing successful": https://github.com/CatalaLang/catala/pull/341/commits/498429e4b7765634c05e538dc96e28524e6e1410#diff-bde130b0c71ac3cb3a70fd86df22d09dc60f2b704391cf2c7bf2fb09c5613dfbL15